bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java

```diff
@@ -159,13 +163,15 @@ public void initialize() {
         String ipAddress = googletvConfig.ipAddress;
         boolean gtvEnabled = googletvConfig.gtvEnabled;
 
-        if (ipAddress.isBlank()) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "@text/offline.googletv-address-not-specified");
-            return;
-        }
+        setThingProperty(PARAMETER_IP_ADDRESS, ipAddress);
```

Comment:
Why do you need to have this as a property as well?

Reply:
That had to be there to fix discovery.  We found the same on Roku recently.  By setting it there, the duplicate entry isn't populated in the inbox.

Reply:
But the parameter is required, so it should be there even for manually created things. Can you share some more details about what is not working? I'm pretty sure this should work.

Reply:
In https://github.com/openhab/openhab-addons/pull/16210 we ended up having to add thing.setProperty(PROPERTY_UUID, deviceInfo.getSerialNumber().toLowerCase()); to RokuHandler.java for the same reason.  If the property is not set, then manually created devices (either GUI or text) are not recognized as the same thing, and thus not auto-ignored.  representation-property only checks the properties, not the parameters on the thing config.  In the case of ATV, there isn't a serial number or anything else that's unique to set to, especially when looking across the protocols.  The only thing I have is the IP of the device.  In this PR we modify the discovery participants to add .withRepresentationProperty(PARAMETER_IP_ADDRESS).withProperty(PARAMETER_IP_ADDRESS, ipAddress).  Reality is, we're just using the same name of "ipAddress" as both a parameter and a property.

Reply:
Can you try to add the representation property to the thing definition? See #12217.

Reply:
It's already there. Has been since the original merge.

Reply:
Sorry, I was on the phone and didn't see it. You are right. However, the case seems very similar to Danfoss Air Unit, and here it works. Definition:
https://github.com/openhab/openhab-addons/blob/444d824a41b9d0e517c35e2736e59ad3aabcbeb8/bundles/org.openhab.binding.danfossairunit/src/main/resources/OH-INF/thing/thing-types.xml#L23-L29

Discovery:
https://github.com/openhab/openhab-addons/blob/444d824a41b9d0e517c35e2736e59ad3aabcbeb8/bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/discovery/DanfossAirUnitDiscoveryService.java#L136-L146

And nothing in the handler except using the parameter. The host doesn't exist as a property. When I delete the ignored entry from the inbox and run a new scan, it's again added and automatically ignored:

![image](https://github.com/openhab/openhab-addons/assets/19519842/71429dd0-cec3-4ceb-9176-0c520e4fc8f7)

I can try to have a look later what's going on since I also have an AndroidTV.

Reply:
Just to be clear,  with Danfoss does it also ignore the result when you create the thing manually text and UI?  That's my problem here.

Reply:
Yes,, this is my thing:
```java
Thing danfossairunit:airunit:a2 "Danfoss HRV" [host="192.168.0.234",
refreshInterval=10,
updateUnchangedValuesEveryMillis=60000]
```

Reply:
Interesting.  I wonder why it works there but ATV and Roku did not.  I'll go grab some screen shots as examples.

Reply:
I assume you have configured exactly the same IP address as discovered and not using a hostname?

Reply:
Apparently screenshots are harder than I expected.  The property isn't unesetting from the database when I install the addon from the current snapshot as opposed to the test jar.

Reply:
> Yes,, this is my thing:
> 
> ```java
> Thing danfossairunit:airunit:a2 "Danfoss HRV" [host="192.168.0.234",
> refreshInterval=10,
> updateUnchangedValuesEveryMillis=60000]
> ```

When you go into the thing in the UI, do you see a host property set as well?

Reply:
> I assume you have configured exactly the same IP address as discovered and not using a hostname?

Yes, identical IPs

Reply:
> When you go into the thing in the UI, do you see a host property set as well?

No.

Reply:
The host property being set here:

![properties](https://github.com/openhab/openhab-addons/assets/53797132/c8025429-6db3-43eb-9025-20607966da9b)

Makes these entries ignored:

![inbox](https://github.com/openhab/openhab-addons/assets/53797132/966a285a-e16e-4a41-9c3c-ae04bc195c88)

To note, there is actually another issue I'm not sure how to resolve with this.  Notice the googletv entries don't auto-ignore but the shieldtv do.  I'm not sure how to make the googletv's tie to the shieldtvs existing.  I'm assuming this would have to be added to core to work.

Reply:
> No.

Interesting.  Not sure why it works on some and not on others then.

Reply:
Ok, I was able to replicate this.

I started with the binding as it exists in the snapshot I'm on (3802 for reference).  I created a thing manually in the UI for my test system.  Notice, no host property is set here.  Also, this is just a GTV not STV thing.

![gtv-commit](https://github.com/openhab/openhab-addons/assets/53797132/159b0dd0-5787-47b3-b961-80a3fdd3f6c3)

The inbox does not ignore the device.

![discover-commit](https://github.com/openhab/openhab-addons/assets/53797132/f00e2b9b-6168-474d-b3d1-93768286ec24)

I then swapped in the jar.  I did not remove the thing or remove the entry from the inbox.  Notice the host property is added.

![gtv-jar](https://github.com/openhab/openhab-addons/assets/53797132/e0bdb1df-b922-498e-9517-277482a62361)

I then rescanned ATV and voila the entry is now ignored.

![discover-jar](https://github.com/openhab/openhab-addons/assets/53797132/62672923-31b6-43c1-846a-60b4ae020437)

I'm not entirely sure of the why, because I'm not that familiar with the code for discovery in the core, but by setting ipAddress as a property it resolves the issue.

Reply:
> To note, there is actually another issue I'm not sure how to resolve with this. Notice the googletv entries don't auto-ignore but the shieldtv do. I'm not sure how to make the googletv's tie to the shieldtvs existing. I'm assuming this would have to be added to core to work.

You could try to return only the thing type UID for ShieldTV here:
https://github.com/openhab/openhab-addons/blob/444d824a41b9d0e517c35e2736e59ad3aabcbeb8/bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/discovery/ShieldTVDiscoveryParticipant.java#L46-L49

Reply:
> I then rescanned ATV and voila the entry is now ignored.

With the snapshot version of the binding, did you also try to delete entries from the inbox to have a clean slate, and then perform a rescan?

Reply:
Also, as a comparison.  Roku creates a property called uuid which effectively the serial number in lower case and adds that as a property both in discovery and in the handler.  The only difference here is that uuid isn't a thing parameter.  I'm not sure if it matters that I've reused ipAddress and they created uuid or not.  Either way, it's the correlation to uuid that's making it ignore.  To test that originally I created a branch and only updated that one single line and it resolved the issue.  See https://github.com/openhab/openhab-addons/issues/16164 for that info.

Reply:
> > To note, there is actually another issue I'm not sure how to resolve with this. Notice the googletv entries don't auto-ignore but the shieldtv do. I'm not sure how to make the googletv's tie to the shieldtvs existing. I'm assuming this would have to be added to core to work.
> 
> You could try to return only the thing type UID for ShieldTV here:
> 
> https://github.com/openhab/openhab-addons/blob/444d824a41b9d0e517c35e2736e59ad3aabcbeb8/bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/discovery/ShieldTVDiscoveryParticipant.java#L46-L49

I will give that a shot and submit a PR if I can get it working.  Thank you!

Reply:
> > I then rescanned ATV and voila the entry is now ignored.
> 
> With the snapshot version of the binding, did you also try to delete entries from the inbox to have a clean slate, and then perform a rescan?
> 

Yes I deleted the entry in the inbox and rescanned to get that first result above (the unignored entry) just to be sure.

Reply:
For uniformity in testing, this is the jar I'm currently running.  

https://github.com/morph166955/openhab-addons/releases/download/androidtv-beta/org.openhab.binding.androidtv-4.2.0-SNAPSHOT.jar

Reply:
> I will give that a shot and submit a PR if I can get it working. Thank you!

Going back to my comment here https://github.com/openhab/openhab-addons/pull/16191#discussion_r1449337947 - perhaps you could do a new PR for both discovery issues and postpone resolving #16223 until we have this figured out? Then this PR can be merged now.

Reply:
Sure. Gimme an hour or two to get to my box and I'll split it apart.